src/Diagnostics.RuntimeHost/Services/StorageService/StorageService.cs

```diff
@@ -81,37 +82,52 @@ public StorageService(IConfiguration configuration, IHostingEnvironment hostingE
 
         public async Task<List<DiagEntity>> GetEntitiesByPartitionkey(string partitionKey = null)
         {
-            try
+            int retryThreshold = 2;
+            int attempt = 0;
```

Comment:
Have you thought about using https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.table.tablerequestoptions.retrypolicy?view=azure-dotnet#Microsoft_Azure_Cosmos_Table_TableRequestOptions_RetryPolicy? I know were not using the cosmos library but I believe this is an older concept in the older libraries too.

Reply:
So there are native implementations of retry policy already in SDK so I did not want to implement anything special unless I find a reason to. https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.storage.blob.blobrequestoptions.retrypolicy?view=azure-dotnet-legacy